Mathlib/Data/List/SplitLengths.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Order.MinMax
+import Mathlib.Algebra.Group.Nat
+
+/-!
+# Splitting a list to chunks of specified lengths
+
+This file defines splitting a list to chunks of given lengths, and some proofs about that.
+-/
+
+variable {α : Type*} (l : List α) (sz : List ℕ)
+
+namespace List
+
+/--
+Split a list to chunks of given lengths.
+-/
+def splitLengths : List ℕ → List α → List (List α)
```

Comment:
Perhaps a minor detail, but I think it'd make more sense for the `List α` argument to go first. That way `l.splitLengths m` is "split `l` according to `m`" rather than the other way around.

Reply:
That's how it was originally, but Kim suggested swapping it because
> Given a list of lengths this is then a nice operator List a \to List (List a). Also the fact that's the way the induction runs!